src/robotFactory.js

```diff
@@ -1,10 +1,84 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    const { x = 0, y = 0 } = coords;
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
 
-class DeliveryDrone {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, `
+    + `Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    const { x = 0, y = 0, z = 0 } = coords;
+
+    super(name, weight, coords, chipVersion);
+
+    this.coords = {
+      x: x,
+      y: y,
+      z: z,
+    };
+  }
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords = {}, chipVersion,
+    maxLoadWeight, currentLoad) {
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (this.currentLoad === undefined || this.currentLoad === null) {
+      if (this.maxLoadWeight >= cargo.weight) {
+        this.currentLoad = cargo;
+      }
```

Comment:
```suggestion
    if (!this.currentLoad === undefined && this.maxLoadWeight >= cargo.weight) {
        this.currentLoad = cargo;
    }
```
Combine these checks in one

Reply:
Тут необхідно також перевірити чи this.currentLoad не є null, тому записано таким чином

Reply:
`if (!this.currentLoad && cargo.weight <= this.maxLoadWeight)`

`!this.currentLoad` will check both null and undefined

Reply:
перероблю, просто раніше мені наголошували що такі перевірки не дуже надійні, бо якщо буде 0, то також повернеться false